pkg/controller/backupstoragelocation_controller.go

```diff
@@ -44,49 +57,62 @@ func (r *BackupStorageLocationReconciler) Reconcile(req ctrl.Request) (ctrl.Resu
 
 	log.Info("Checking for existing backup locations ready to be verified; there needs to be at least 1 backup location available")
 
-	locationList, err := velero.ListBackupStorageLocations(r.StorageLocation.Client, r.StorageLocation.Ctx, req.Namespace)
+	locationList, err := storage.ListBackupStorageLocations(r.Ctx, r.Client, req.Namespace)
 	if err != nil {
 		log.WithError(err).Error("No backup storage locations found, at least one is required")
 		return ctrl.Result{Requeue: true}, err
 	}
 
+	pluginManager := r.NewPluginManager(log)
+	defer pluginManager.CleanupClients()
+
 	var defaultFound bool
 	var unavailableErrors []string
 	var anyVerified bool
 	for i := range locationList.Items {
 		location := &locationList.Items[i]
 		log := r.Log.WithField("controller", "backupstoragelocation").WithField("backupstoragelocation", location.Name)
 
-		if location.Name == r.StorageLocation.DefaultStorageLocation {
+		if location.Name == r.DefaultBackupLocationInfo.StorageLocation {
 			defaultFound = true
 		}
 
-		if !r.StorageLocation.IsReadyToValidate(location, log) {
+		backupStore, err := r.NewBackupStore(location, pluginManager, log)
+		if err != nil {
+			log.WithError(err).Error("Error getting a backup store")
 			continue
 		}
 
-		anyVerified = true
+		if !storage.IsReadyToValidate(location.Spec.ValidationFrequency, location.Status.LastValidationTime, r.DefaultBackupLocationInfo, log) {
+			log.Debug("Backup location not ready to be validated")
+			continue
+		}
 
-		log.Debug("Verifying backup storage location")
+		// Initialize the patch helper.
+		patchHelper, err := patch.NewHelper(location, r.Client)
```

Comment:
We should set `patchHelper.isConditionsSetter` explicitly as it involves code specific to CAPI.
https://github.com/kubernetes-sigs/cluster-api/blob/master/util/patch/patch.go#L155

Though there is this https://github.com/kubernetes-sigs/cluster-api/blob/master/util/patch/patch.go#L64
```
	// Check if the object satisfies the Cluster API conditions contract.
	_, canInterfaceConditions := obj.(conditions.Setter)
```
IMO, it is better to set this to false explicitly.

Reply:
Not sure about this suggestion. What value would this be adding? The check on the capi side will always happen, and it's clutter that whoever is reading this code doesn't need to reason thru, unless there's value to doing this that I'm not seeing.

Reply:
Running the check on https://github.com/kubernetes-sigs/cluster-api/blob/master/util/patch/patch.go#L64 is not a problem. However, based on whether `isConditionsSetter` is true or false, the code branches taken by `patchHelper.Patch` is different. The code branch taken when `isConditionsSetter` is true, is specific to CAPI which should not be called when patching Velero resource.

Reply:
Right. And we are not setting it. This means that branch will not validate to `true`.

Reply:
In other words, I know what the code does (I've read it), what I am saying is that I don't agree that making this explicit makes this code better, I think it makes it worse, since this is would be unnecessary code added.